src/spaceone/inventory/service/cloud_service_service.py

```diff
@@ -181,30 +176,42 @@ def update(self, params):
         old_cloud_svc_data = dict(cloud_svc_vo.to_dict())
 
         if 'tags' in params:
-            new_tags = self._create_tag(params, tag_type)
+            old_tags = old_cloud_svc_data.get('tags', {})
+            new_tags = self._create_tags(params['tags'], provider)
+            old_tag_keys = old_cloud_svc_data.get('tag_keys', {})
+            new_tag_keys = self._create_tag_keys(params['tags'], provider)
+
+            if new_tags != old_tags:
+                old_tags.update(new_tags)
+                old_tag_keys.update(new_tag_keys)
+                params['tags'] = old_tags
+                params['tag_keys'] = old_tag_keys
```

Comment:
Declare a set to remove duplicate keys.
`params['tag_keys'] = list(set(old_tags_keys))`